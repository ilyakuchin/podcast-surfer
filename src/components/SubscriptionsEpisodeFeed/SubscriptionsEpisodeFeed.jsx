import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchFeed } from '../../redux/actions/Feed/feed';
import { EpisodesList } from '../EpisodesList/EpisodesList';
import { setCurrentEpisode } from '../../redux/actions/CurrentEpisode/currentEpisode';

export function SubscriptionsEpisodeFeed({
  fetchFeedConnect,
  feed,
  subscriptions,
  setCurrentEpisodeConnect,
  isFetching,
}) {
  useEffect(() => fetchFeedConnect(subscriptions), [
    fetchFeedConnect,
    subscriptions,
  ]);

  function podcastGrid() {
    return !feed.length > 0 ? (
      <h2>Feed is empty</h2>
    ) : (
      <div>
        <h2>Feed</h2>
        <div>
          <EpisodesList
            episodes={feed}
            setCurrentEpisodeConnect={setCurrentEpisodeConnect}
          />
        </div>
      </div>
    );
  }

  return !isFetching ? podcastGrid() : <div>Loading...</div>;
}

SubscriptionsEpisodeFeed.propTypes = {
  fetchFeedConnect: PropTypes.func.isRequired,
  feed: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      imageUrl: PropTypes.string,
    })
  ).isRequired,
  subscriptions: PropTypes.arrayOf(PropTypes.string).isRequired,
  setCurrentEpisodeConnect: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => {
  return {
    subscriptions: state.userInfo.subscriptions,
    feed: state.feed.feed,
    isFetching: state.feed.isFetching,
  };
};

const dispatchToProps = (dispatch) => {
  return {
    fetchFeedConnect: (urlList) => dispatch(fetchFeed(urlList)),
    setCurrentEpisodeConnect: (name, description, imageUrl, audioUrl) =>
      dispatch(setCurrentEpisode(name, description, imageUrl, audioUrl)),
  };
};

const ConnectedSubscriptionsEpisodeFeed = connect(
  mapStateToProps,
  dispatchToProps
)(SubscriptionsEpisodeFeed);

export default ConnectedSubscriptionsEpisodeFeed;
