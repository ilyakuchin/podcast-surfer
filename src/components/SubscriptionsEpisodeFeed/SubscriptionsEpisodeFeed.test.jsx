import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SubscriptionsEpisodeFeed } from './SubscriptionsEpisodeFeed';

configure({ adapter: new Adapter() });

describe('Test SubscriptionsEpisodeFeed component', () => {
  test('should render correctly', () => {
    expect(
      toJson(
        shallow(
          <SubscriptionsEpisodeFeed
            fetchFeedConnect={() => {}}
            feed={[]}
            isFetching={false}
            subscriptions={[]}
            setCurrentEpisodeConnect={() => {}}
          />
        )
      )
    ).toMatchSnapshot();
  });

  test('should render correctly', () => {
    expect(
      toJson(
        shallow(
          <SubscriptionsEpisodeFeed
            isFetching
            fetchFeedConnect={() => {}}
            feed={[]}
            subscriptions={[]}
            setCurrentEpisodeConnect={() => {}}
          />
        )
      )
    ).toMatchSnapshot();
  });
});
