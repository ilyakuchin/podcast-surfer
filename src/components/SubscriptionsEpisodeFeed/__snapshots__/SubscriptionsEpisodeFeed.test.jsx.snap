// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SubscriptionsEpisodeFeed component should render correctly 1`] = `
<h2>
  Feed is empty
</h2>
`;

exports[`Test SubscriptionsEpisodeFeed component should render correctly 2`] = `
<div>
  Loading...
</div>
`;
